protocols/ping/src/dial_handler.rs

```diff
@@ -232,100 +229,92 @@ where
             )
         }
 
-        loop {
```

Comment:
Curious, what was the error reported here? Was the loop never looped, i.e. the code always returned?

Reply:
> Was the loop never looped [...]

Yes.